BLSQ/iaso
# Allow pagination of mobile OrgUnit API

We received a complaint that some devices cannot synchronize OrgUnits when they are too many.
As we couldn't reproduce, the assumptions are:
 - The list is too big in memory
 - The filtering that is done to remove the `parent_id` value of children for which the parent hasn't been returned is too consuming.

This PR allows 2 new parameters: `page_size`and `page`, to process the OrgUnit in smaller chunks.
Since we won't be able to remove the `parent_id` of children without parents with only part of the data, the removal is now done in the backend.

Therefore, this addresses both assumptions while still being backward compatible.

Related JIRA tickets: IA-1914

## Self proofreading checklist


## Changes

The `/api/mobile/orgunits` endpoint is now capable of returning a paginated list.
The `parent_id` value of OrgUnits with a parent that has a `REJECTED` status is now set to `None`.

## How to test

Call `/api/mobile/orgunits?app_id=...` **before** applying the changes.
Call `/api/mobile/orgunits?app_id=...` **after** applying the changes.
The results should be the same

Then, call `/api/mobile/orgunits?app_id=...` with a `page_size` and `page` parameters and ensure the right chunk of the data has been returned.
